spec/controllers/cycle_counts_controller_spec.rb

```diff
@@ -0,0 +1,202 @@
+      it_behaves_like 'an edit request'
+      it { expect(send(:cycle_count)).to be_kind_of(CycleCount) }
+    end
+
+    context 'when not authorized' do
+      let(:authorized?) { raise_pundit_error CycleCountPolicy }
+
+      it_behaves_like 'an unauthorized access to a resource'
+    end
+  end
+
+  describe 'PATCH update' do
+    before do
+      allow(cycle_count).to receive(:update_attributes) { update? }
+      patch :update,
+            id: cycle_count.id,
+            cycle_count: attributes_for(:cycle_count)
+    end
+
+    let(:update?) { fail 'update? not set' }
+
+    context 'when the call is successful' do
+      let(:authorized?) { true }
+      let(:update?) { true }
+
+      it_behaves_like 'a redirect'
+      it { is_expected.to redirect_to(cycle_count_path(cycle_count)) }
+      it do
+        is_expected
+          .to set_flash[:notice]
+          .to I18n.t('cycle_counts.update.success')
+      end
+    end
+
+    context 'when the call is not successful' do
+      let(:authorized?) { true }
+      let(:update?) { false }
+
+      it_behaves_like 'a successful request'
+      it_behaves_like 'an edit request'
+      it do
+        is_expected
+          .to set_flash[:alert]
+          .to(I18n.t('cycle_counts.update.failure'))
+      end
+    end
+
+    context 'when not authorized' do
+      let(:authorized?) { raise_pundit_error CycleCountPolicy }
+      let(:cycle_count_params) { nil }
+
+      it_behaves_like 'an unauthorized access to a resource'
+    end
+  end
+
+  describe 'POST create' do
+    before do
+      allow(cycle_count).to receive(:save) { create? }
+      allow(CycleCount).to receive(:new) { cycle_count }
+
+      post :create, cycle_count: nil
+    end
+
+    let(:cycle_count) { build_stubbed(:cycle_count) }
+    let(:create?) { fail 'create? not set' }
+
+    context 'when the call is successful' do
+      let(:authorized?) { true }
+      let(:create?) { true }
+
+      it_behaves_like 'a redirect'
+      it { is_expected.to redirect_to(cycle_count_path(assigns(:cycle_count))) }
+      it do
+        is_expected
+          .to set_flash[:notice]
+          .to I18n.t('cycle_counts.create.success')
+      end
+    end
+
+    context 'when the call is not successful' do
+      let(:authorized?) { true }
+      let(:create?) { false }
+
+      it_behaves_like 'a successful request'
+      it_behaves_like 'a new request'
+      it do
+        is_expected
+          .to set_flash[:alert]
+          .to(I18n.t('cycle_counts.create.failure'))
+      end
+    end
+
+    context 'when not authorized' do
+      let(:authorized?) { raise_pundit_error CycleCountPolicy }
+      let(:cycle_count_params) { nil }
+
+      it_behaves_like 'an unauthorized access to a resource'
+    end
+  end
+
+  describe 'DELETE destroy' do
+    before do
+      allow(cycle_count).to receive(:destroy) { destroy? }
+      allow(CycleCount).to receive(:find) { cycle_count }
+
+      xhr :delete, :destroy, id: cycle_count.id
```

Comment:
Because I copied one of your delete tests instead of writing it myself.  I'll change it.